### PR TITLE
CARTO: Add example for boundaries

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/8148

#### Change List

- Add `CartoPropertiesTileLoader` to load binary property tile data
- Enhance `VectorTileLayer` to support fetching & merging of property data
- Add `cartoBoundaryQuerySource` & `cartoBoundaryTableSource`
- Update test app with query & table boundary examples

### CODE REVIEW

**Review by alasarr — CHANGES_REQUESTED** [review]

File: modules/carto/src/sources/boundary-query-source.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import {baseSource} from './base-source';
```
[the remark this answers is not in the document]
Comment (felixpalmer) [reply modules/carto/src/sources/boundary-query-source.ts:1]: As discussed, it's better to be explicit & consistent and add a `boundaryTilesetSource` which has no such property

File: modules/carto/src/sources/boundary-query-source.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import type {SourceOptions, TilejsonResult} from './types';
+
+export type BoundaryQuerySourceOptions = SourceOptions & {
+  boundaryId: string;
+  matchingPropertyName?: string;
+  propertiesSqlQuery: string;
```
Comment (alasarr) [inline modules/carto/src/sources/boundary-query-source.ts:7]: rename to sqlQuery

File: modules/carto/src/sources/boundary-table-source.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import {baseSource} from './base-source';
+import type {SourceOptions, TilejsonResult} from './types';
+
+export type BoundaryTableSourceOptions = SourceOptions & {
+  boundaryId: string;
+  columns?: string[];
+  matchingPropertyName?: string;
+  propertiesTableName: string;
```
Comment (alasarr) [inline modules/carto/src/sources/boundary-table-source.ts:8]: tableName

File: modules/carto/src/sources/boundary-table-source.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import {baseSource} from './base-source';
+import type {SourceOptions, TilejsonResult} from './types';
+
+export type BoundaryTableSourceOptions = SourceOptions & {
+  boundaryId: string;
+  columns?: string[];
+  matchingPropertyName?: string;
```
Comment (alasarr) [inline modules/carto/src/sources/boundary-table-source.ts:7]: matchingColumn

File: docs/api-reference/carto/data-sources.md [hunk 5]
```diff
@@ -147,19 +144,50 @@ type quadbinQuerySourceOptions = {
...
+  queryParameters: QueryParameters;
+}
+```
+
+#### boundaryTilesetSource
+
+```ts
+type BoundaryTilesetSourceOptions = {
+  boundaryId: string;
+  columns?: string[];
```
Comment (alasarr) [inline docs/api-reference/carto/data-sources.md:187]: Why is this required?  In tileset we don't support columns as the API cannot filter by them

File: modules/carto/src/sources/boundary-tileset-source.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  columns?: string;
+};
+
+export const boundaryTilesetSource = async function (
+  options: BoundaryTilesetSourceOptions
+): Promise<TilejsonResult> {
+  const {boundaryId, columns} = options;
+  const urlParameters: UrlParameters = {boundaryId};
+
+  if (columns) {
```
Comment (alasarr) [inline modules/carto/src/sources/boundary-tileset-source.ts:19]: Not required

**Review by alasarr — CHANGES_REQUESTED** [review]
Just a small thing detected

**Review by alasarr — APPROVED** [review]

File: docs/api-reference/carto/data-sources.md [hunk 7]
```diff
@@ -147,19 +144,50 @@ type quadbinQuerySourceOptions = {
...
 #### rasterTilesetSource
 
 ```ts
-type rasterTilesetSourceOptions = {
+type RasterTilesetSourceOptions = {
+  tableName: string;
+}
+```
+
+#### boundaryTableSource
```
Comment (alasarr) [inline docs/api-reference/carto/data-sources.md:160]: We should mark this as experimental

**Pull request merged**